src/symbolize/gimli/elf.rs

```diff
@@ -102,14 +102,14 @@ impl Mapping {
 
             // Try to locate a supplementary object file.
             let mut sup = None;
-            if let Some((path_sup, build_id_sup)) = object.gnu_debugaltlink_path(&path) {
-                if let Some(map_sup) = super::mmap(&path_sup) {
-                    let map_sup = stash.cache_mmap(map_sup);
-                    if let Some(sup_) = Object::parse(map_sup) {
-                        if sup_.build_id() == Some(build_id_sup) {
-                            sup = Some(sup_);
-                        }
-                    }
+            if let Some((path_sup, build_id_sup)) = object.gnu_debugaltlink_path(&path)
+                && let Some(map_sup) = super::mmap(&path_sup)
+            {
+                let map_sup = stash.cache_mmap(map_sup);
```

Comment:
I think I want a better code pattern for this particular motion, since we do it in many places.